sentry/src/test/java/io/sentry/metrics/MetricsApiTest.kt

```diff
@@ -0,0 +1,212 @@
+package io.sentry.metrics
+
+import io.sentry.IMetricsAggregator
+import io.sentry.metrics.MetricsApi.IMetricsInterface
+import org.mockito.kotlin.any
+import org.mockito.kotlin.anyOrNull
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import kotlin.test.Test
+
+class MetricsApiTest {
+
+    @Test
+    fun `default timestamp is provided`() {
+        val aggregator = mock<IMetricsAggregator>()
+        val api = MetricsApi(object : IMetricsInterface {
+            override fun getMetricsAggregator(): IMetricsAggregator {
+                return aggregator
+            }
+
+            override fun getDefaultTagsForMetrics(): Map<String, String> = emptyMap()
+        })
+
+        api.increment("name", 1.0, null, null, null)
+        api.set("name", 1, null, null, null)
+        api.set("name", "string", null, null, null)
+        api.gauge("name", 1.0, null, null, null)
+        api.distribution("name", 1.0, null, null, null)
+
+        verify(aggregator).increment(
+            anyOrNull(),
+            anyOrNull(),
+            anyOrNull(),
+            anyOrNull(),
+            any(),
+            anyOrNull()
+        )
+
+        verify(aggregator).set(
+            anyOrNull(),
+            eq(1),
+            anyOrNull(),
+            anyOrNull(),
+            any(),
+            anyOrNull()
+        )
+
+        verify(aggregator).set(
+            anyOrNull(),
+            eq("string"),
+            anyOrNull(),
+            anyOrNull(),
+            any(),
+            anyOrNull()
+        )
+
+        verify(aggregator).gauge(
+            anyOrNull(),
+            anyOrNull(),
+            anyOrNull(),
+            anyOrNull(),
+            any(),
+            anyOrNull()
+        )
+
+        verify(aggregator).distribution(
+            anyOrNull(),
+            anyOrNull(),
+            anyOrNull(),
+            anyOrNull(),
+            any(),
+            anyOrNull()
+        )
+    }
+
+    @Test
+    fun `timestamp is not overwritten`() {
```

Comment:
shouldn't this be `timestamp is overwritten`?

Reply:
No, it's like this: when the timestamp is provided the `MetricsApi` shouldn't overwrite it.